src/api/c/vector_field.cpp

```diff
@@ -50,25 +51,23 @@ fg_chart setup_vector_field(fg_window window, const vector<af_array>& points,
     vector<Array<T>> pnts;
     vector<Array<T>> dirs;
 
+    dim_t ndots = 0;
     for (unsigned i = 0; i < points.size(); ++i) {
         pnts.push_back(getArray<T>(points[i]));
         dirs.push_back(getArray<T>(directions[i]));
+        ndots += getInfo(points[i]).dims().dims[0];
     }
 
     // Join for set up vector
-    dim4 odims(3, points.size());
-    Array<T> out_pnts = createEmptyArray<T>(odims);
-    Array<T> out_dirs = createEmptyArray<T>(odims);
-    detail::join(out_pnts, 1, pnts);
-    detail::join(out_dirs, 1, dirs);
```

Comment:
> Uninitialized buffer used to collect partial join (2D -> 3D)

It has been a while I looked at this section. Can you please elaborate since you have been looking at this recently. The dimensions of the arrays are identical here and join writes to output arrays, overwriting uninitialized values.

Reply:
Sorry for the late response, I missed this question.

Depending on the request for transpose or not, the dimensions are different.
With transpose == true --> [N,1,1] + [N,1,1] + [N,1,1] --> [N,3,1]
With transpose == false --> [1,N,1] + [1,N,1] + [1,N,1] --> [3,N,1]
Remark: the order of the elements in the data buffer is the same for both options.  The dimensions are however different.

In the old implementation of the join methods, the dimensions were not verified with the risk of "out of buffer" writes, with unpredictable results as a consequence.  In many cases the reserve of an oversized buffer is used (unnoticed) or the next array in memory is overwritten (unpredictable effect).